commerce_coordinator/apps/commercetools/pipeline.py

```diff
@@ -70,25 +69,82 @@ def run_filter(self, params, order_data):  # pylint: disable=arguments-differ
             return PipelineCommand.CONTINUE.value
 
 
-class FetchOrderDetails(PipelineStep):
+class FetchOrderDetailsByOrderNumber(PipelineStep):
     """ Fetch the order Details and if we can, set the PaymentIntent """
 
     def run_filter(self, active_order_management_system, order_number, **kwargs):  # pylint: disable=arguments-differ
         """
         Execute a filter with the signature specified.
         Arguments:
-            active_order_management_system: The Active Order System (optional)
-            params: arguments passed through from the original order history url querystring
-            order_number: Order number (for now this is an order.id, but this should change in the future)
-            TODO: SONIC-277 (in-progress)
+            active_order_management_system: The Active Order System
+            order_number: Order number
+            kwargs: The keyword arguments passed through from the filter
+        Returns:
+            order_data (CTOrder): The object of the order
+            payment_intent_id (str): The Stripe PaymentIntent ID
+            amount_in_cents (decimal): Total amount to refund
+            has_been_refunded (bool): Has this payment been refunded
+            payment_data (CTPayment): Payment object of the order
+        """
+        if active_order_management_system != COMMERCETOOLS_ORDER_MANAGEMENT_SYSTEM:
+            return PipelineCommand.CONTINUE.value
+
+        try:
+            ct_api_client = CommercetoolsAPIClient()
+            ct_order = ct_api_client.get_order_by_number(order_number=order_number)
+
+            ret_val = {
+                "order_data": ct_order,
+            }
+
+            intent_id = get_edx_payment_intent_id(ct_order)
+
+            if intent_id:
+                ct_payment = ct_api_client.get_payment_by_key(intent_id)
+                ret_val['payment_intent_id'] = intent_id
+                ret_val['amount_in_cents'] = get_edx_refund_amount(ct_order)
+                ret_val['has_been_refunded'] = has_refund_transaction(ct_payment)
+                ret_val['payment_data'] = ct_payment
+            else:
+                ret_val['payment_intent_id'] = None
+                ret_val['amount_in_cents'] = decimal.Decimal(0.00)
+                ret_val['has_been_refunded'] = False
+                ret_val['payment_data'] = None
+
+            return ret_val
+        except CommercetoolsError as err:  # pragma no cover
+            log.exception(f"[{type(self).__name__}] Commercetools Error: {err}, {err.errors}")
+            return PipelineCommand.CONTINUE.value
+        except HTTPError as err:
+            log.exception(f"[{type(self).__name__}] HTTP Error: {err}")
+            return PipelineCommand.CONTINUE.value
+
+
+class FetchOrderDetailsByOrderID(PipelineStep):
```

Comment:
Wasnt the idea to use order number everywhere thus gutting all use of order ID as input? We may still need it for lower logic but it can be provided by fetching the order by number and adding it to the pipe as a return.